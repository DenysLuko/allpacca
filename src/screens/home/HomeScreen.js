import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import { connect } from 'react-redux'

import { getUserAction } from '../../actions'
import { Button } from '../../components/Button'

import {
  styles
} from '../../styles'

const homeScreenStyles = StyleSheet.create({
  containerComponent: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonsContainer: {
    height: 250,
    top: 50,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  heading: styles.text.headingText
})

export class _HomeScreen extends Component {
  static navigationOptions = {
    title: 'Home',
  }

  render() {
    const {
      user: {
        name
      }
    } = this.props
    console.log(this.props)

    const greeting = name ? `Hello ${name}!` : 'Hello there!'

    return (
      <View style={homeScreenStyles.containerComponent}>
        <Text style={homeScreenStyles.heading}>{greeting}</Text>
        <View style={homeScreenStyles.buttonsContainer}>
          <Button
            type={'homeButton'}
            title="Search"
            onPress={() => this.props.navigation.navigate('SearchScreen')}
          />
          <Button
            type={'homeButton'}
            title="New Journey"
            onPress={() => this.props.navigation.navigate('ProfileScreen')}
          />
          <Button
            type={'homeButton'}
            title="New Delivery"
            onPress={() => this.props.navigation.navigate('MessagesScreen')}
          />
          <Button
            type={'homeButton'}
            title="Get User"
            onPress={() => this.props.getUser()}
          />
        </View>
      </View>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user
})

const mapDispatchToProps = dispatch => ({
  getUser: () => dispatch(getUserAction())
})

export const HomeScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(_HomeScreen)
