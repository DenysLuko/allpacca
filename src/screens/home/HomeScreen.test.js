/**
 * @format
 */

import 'react-native'
import React from 'react'
import { HomeScreen } from './HomeScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const subject = renderer.create(<HomeScreen />).toJSON()
  expect(subject).toMatchSnapshot()
})
