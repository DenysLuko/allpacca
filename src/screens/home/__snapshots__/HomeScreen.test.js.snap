// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-Light",
        "fontSize": 24,
      }
    }
  >
    AllPacca
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 250,
        "justifyContent": "space-between",
        "top": 50,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#4db8b8",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 70,
          "maxWidth": 300,
          "width": "70%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006b6b",
              "fontFamily": "Avenir-Light",
              "fontSize": 24,
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#4db8b8",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 70,
          "maxWidth": 300,
          "width": "70%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006b6b",
              "fontFamily": "Avenir-Light",
              "fontSize": 24,
            }
          }
        >
          New Journey
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#4db8b8",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 70,
          "maxWidth": 300,
          "width": "70%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006b6b",
              "fontFamily": "Avenir-Light",
              "fontSize": 24,
            }
          }
        >
          New Delivery
        </Text>
      </View>
    </View>
  </View>
</View>
`;
